.github/workflows/update-priority.yml

```diff
@@ -0,0 +1,63 @@
+# Updates the Priority field on a project board when a [Pri] label gets added to an issue.
+name: Update Priority Field
+
+# Triggers the workflow when a label is added to an issue.
+on:
+  issues:
+    types:
+      - labeled
+
+# Details needed to update a project field.
+env:
+  LABEL_PREFIX: "[Pri] "
+  PROJECT_NUMBER: 391
+  PROJECT_FIELD: "Priority"
+  ORGANIZATION: "Automattic"
+
+jobs:
+
+  # Updates the Priority field with the appropriate priority level.
+  update_priority: 
+
+    runs-on: ubuntu-latest
+
+    # Only runs if a priority label is added.
+    if:  contains(github.event.label.name, '[Pri]')
+
+    steps:
+    
+      # Generate a token for the GitHub App, which has special permissions to update projects.
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PEM }} 
+
+      # Read the priority label.
+      - name: Get priority
+        id: get_priority
+        env:
+          PRIORITY_LABEL: ${{github.event.label.name}}
+
+        # Store the priority without the label prefix.
+        run: |
+          prefix="${{env.LABEL_PREFIX}}"
+          label="${{github.event.label.name}}"
+          priority=${label#"$prefix"}
+          echo "PRIORITY_LABEL=$priority" >> $GITHUB_ENV
+          echo $PRIORITY_LABEL
+
+      # Update the project field with the priority.
+      # Note: if this step fails, the issue is most likely not in the project.
```

Comment:
**Suggestion (maybe blocking?)**: Bahhh, this is tough... But I think we may want to avoid this if at all possible. Having steps fail in a non-meaningful way (or in a way that could be predicted and avoided) is probably a state we want to avoid. 

I think our best bet is having the first step in this workflow be an API call that checks and evaluates if the current issue is on our project. To do that, we can use either `curl` or the `gh` CLI (as they do in the [sample docs](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions)). Then, all the other steps would only run if the output of that first step is "true" or "1" or something like that.

The API to check that is GraphQL and I know it can be a bit mind-numbing if you're not experienced with that API design (like meeeeee lolol 😅 ), but I think it's worth it in this case!

Let me know how I can help with that -- I'd be glad to go dig through the API docs and try to spin up that checking step! 👍

Reply:
> I think our best bet is having the first step in this workflow be an API call that checks and evaluates if the current issue is on our project.

Totally agree! I had an old attempt at this from yesterday that failed miserably haha. I'll try to resurrect it and see what went wrong. Thanks for your suggestion!